src/App.js

```diff
@@ -1,16 +1,106 @@
 import React from 'react';
+import TodoForm from './components/TodoComponents/TodoForm';
+import TodoList from './components/TodoComponents/TodoList';
+
+import './App.css';
+
+const todoListData = [
+	{
+		task: 'Organize Garage',
+		id: 12,
+		completed: false
+	},
+	{
+		task: 'Bake Cookies',
+		id: 1528817084358,
+		completed: false
+	},
+	{
+		task: 'Bake Cookies',
+		id: 15288170843585,
+		completed: false
+	}
+];
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
-  render() {
-    return (
-      <div>
-        <h2>Welcome to your Todo App!</h2>
-      </div>
-    );
-  }
+	constructor() {
+		super();
+		this.state = {
+			todos: todoListData,
+			task: '',
+			search: ''
+		};
+	}
+
+	// Search start ------------------------------------------
+
+	handleSearch = (event) => {
+		this.setState({
+			todos: this.state.todos.filter((item) => {
+				return item.task.includes(event.target.value);
+			})
+		});
+	};
+
+	handleChanges = (event) => {
+		this.setState({ [event.target.name]: event.target.value });
+	};
+
+	addNew = (event) => {
+		event.preventDefault();
+		const newTask = [ ...this.state.todos, { task: this.state.task, id: Date.now(), completes: false } ];
+		localStorage.setItem('todo', JSON.stringify(newTask)); //------Storage---------------------------------------------------
+		this.setState({
+			todos: newTask,
+			task: ''
+		});
+	};
+
+	toggleCompleted = (id) => {
+		this.setState({
+			todos: this.state.todos.map((todo) => {
```

Comment:
`setState() does not always immediately update the component. It may batch or defer the update until later. This makes reading this.state right after calling setState() a potential pitfall.`